sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ColumnPruningSuite.scala

```diff
@@ -35,7 +35,8 @@ class ColumnPruningSuite extends PlanTest {
   object Optimize extends RuleExecutor[LogicalPlan] {
     val batches = Batch("Column pruning", FixedPoint(100),
       ColumnPruning,
-      CollapseProject) :: Nil
+      CollapseProject,
+      PushPredicateThroughProject) :: Nil
```

Comment:
In the batch of `Optimizer`, `PushPredicateThroughProject` is called before `ColumnPruning`. Different orders could hide the potential issue.